utils/src/main/scala/za/co/absa/enceladus/utils/unicode/ParameterConversion.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.utils.unicode
+
+object ParameterConversion {
+
+  implicit class UnicodeValueString(s: String) {
+    val unicodeElement = """[uU]?\+?([0-9a-fA-F]{4})""".r
```

Comment:
there is a POSIX class `\p{XDigit}`. What are our thoughts on using that? It is pre-defined and has rules that can be easily referenced.?

Reply:
Hey, how widely is this supported? Can I try this on https://regex101.com/ ?

Reply:
I'd also add an explicit end of string so that '0AA01' won't match as '0AA0'.
```suggestion
    val unicodeElement = """[uU]?\+?([0-9a-fA-F]{4})$""".r
```

Reply:
> Hey, how widely is this supported? Can I try this on https://regex101.com/ ?

I think you cannot. It is a POSIX class, meaning that bash, scala, and other languages will support it. Regex101 I think has no such classes pre-defined.